zenoh/src/net/routing/hat/linkstate_peer/pubsub.rs

```diff
@@ -605,8 +605,28 @@ impl HatPubSubTrait for HatCode {
         }
     }
 
-    fn get_subscriptions(&self, tables: &Tables) -> Vec<Arc<Resource>> {
-        hat!(tables).peer_subs.iter().cloned().collect()
+    fn get_subscriptions(&self, tables: &Tables) -> Vec<(Arc<Resource>, Sources)> {
+        hat!(tables)
+            .peer_subs
+            .iter()
+            .map(|s| {
+                (
+                    s.clone(),
+                    Sources {
```

Comment:
Please add comments and what's happening here. The higher-level logic is not clear from this code.